diff --git a/sources/content/src/ContentEntitySourcePluginUi.php b/sources/content/src/ContentEntitySourcePluginUi.php
index caf6c52..ccbe983 100644
--- a/sources/content/src/ContentEntitySourcePluginUi.php
+++ b/sources/content/src/ContentEntitySourcePluginUi.php
@@ -34,8 +34,8 @@ class ContentEntitySourcePluginUi extends SourcePluginUiBase {
     $field_definitions = \Drupal::entityManager()->getBaseFieldDefinitions($type);
 
     $label_key = $entity_type->getKey('label');
-    if (!empty($label_key)) {
-      $label = (string) $field_definitions[$label_key]->getlabel();
+    if (!empty($label_key) && !empty($field_definitions[$label_key])) {
+      $label = (string) $field_definitions[$label_key]->getLabel();
       $form['search_wrapper']['search'][$label_key] = array(
         '#type' => 'textfield',
         '#title' => $label,
